cmd/kubelet/app/server.go

```diff
@@ -286,6 +292,114 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
+		}
+	}
+
+	return kubeletConfigE, kubeletConfigInstanceE, nil
+}
+
+// loadInternalConfigurations converts the external shared and instance configuration to their respective
+func loadConfiguration(configFile, configInstanceFile string, args []string) (*kubeletconfiginternal.KubeletConfiguration, *kubeletconfiginternal.KubeletInstanceConfiguration, error) {
+	kubeletConfigE, kubeletConfigInstanceE, err := loadExternalConfiguration(configFile, configInstanceFile)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// convert external versioned configuration in internal
+	kubeletConfig, err := utilcodec.ConvertKubeletConfigurationI(kubeletConfigE)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// convert external instance versioned configuration in internal
+	kubeletInstanceConfig, err := utilcodec.ConvertKubeletConfigurationInstanceI(kubeletConfigInstanceE)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// make all paths absolute
+	utilkubeletconfig.ResolveRelativePaths(
+		kubeletconfiginternal.KubeletConfigurationPathRefs(kubeletConfig),
+		filepath.Dir(configFile),
+	)
+
+	// We must enforce flag precedence by re-parsing the command line into the new object.
+	// This is necessary to preserve backwards-compatibility across binary upgrades.
+	// See issue #56171 for more details.
+	if err := kubeletConfigFlagPrecedence(kubeletConfig, kubeletInstanceConfig, args); err != nil {
+		return nil, nil, err
+	}
+
+	return kubeletConfig, kubeletInstanceConfig, nil
+}
+
+// mergeKubeletConfigurations merges the fields from KubeletInstanceConfiguration to KubeletConfiguration
+func mergeKubeletConfigurationI(
+	kc *kubeletconfiginternal.KubeletConfiguration,
+	kic *kubeletconfiginternal.KubeletInstanceConfiguration,
+) {
+	if kc.Address != kic.Address {
+		kc.Address = kic.Address
+	}
+}
+
+// mergeKubeletConfigurations merges the fields from KubeletInstanceConfiguration to KubeletConfiguration
+func mergeKubeletConfigurationE(
+	kc *kubeletconfigv1beta1.KubeletConfiguration,
+	kic *kubeletconfigv1beta1.KubeletInstanceConfiguration,
+) {
+	if kc.Address != kic.Address {
+		kic.Address = kc.Address
+	}
+}
+
+// loadDynamicConfiguration returns the Last Known Good configuration and the Kubelet config controller
+func loadDynamicConfiguration(dynamicConfigDir string, args []string) (*kubeletconfiginternal.KubeletConfiguration, *dynamickubeletconfig.Controller, error) {
+	dynamicKubeletConfig, kubeletConfigController, err := BootstrapKubeletConfigController(dynamicConfigDir,
+		func(kc *kubeletconfiginternal.KubeletConfiguration) error {
+			// Here, we enforce flag precedence inside the controller, prior to the controller's validation sequence,
+			// so that we get a complete validation at the same point where we can decide to reject dynamic config.
+			// This fixes the flag-precedence component of issue #63305.
+			// See issue #56171 for general details on flag precedence.
+			return kubeletConfigFlagPrecedence(kc, nil, args)
```

Comment:
Pass `&KubeletInstanceConfiguration{}` instead, rather than conditionally registering flags (missing flags will cause the command line to fail to parse, better we always require they be registered).